examples/date_binding.h

```diff
@@ -0,0 +1,127 @@
+ *  This is where magic happens. To tell sqlite_orm how to act
+ *  with SysDays we have to create a few service classes
+ *  specializations (traits) in sqlite_orm namespace.
+ */
+namespace sqlite_orm {
+
+    /**
+	 *  First of all is a type_printer template class.
+	 *  It is responsible for sqlite type string representation.
+	 *  We want SysDays to be `TEXT` so let's just derive from
+	 *  text_printer. Also there are other printers: real_printer and
+	 *  integer_printer. We must use them if we want to map our type to `REAL` (double/float)
+	 *  or `INTEGER` (int/long/short etc) respectively.
+	 */
+    template<>
+    struct type_printer<std::chrono::sys_days> : public text_printer {};
+
+    /**
+	 *  This is a binder class. It is used to bind c++ values to sqlite queries.
+	 *  Here we have to create sysday string representation and bind it as string.
+	 *  Any statement_binder specialization must have `int bind(sqlite3_stmt*, int, const T&)` function
+	 *  which returns bind result. Also you can call any of `sqlite3_bind_*` functions directly.
+	 *  More here https://www.sqlite.org/c3ref/bind_blob.html
+	 */
+    template<>
+    struct statement_binder<std::chrono::sys_days> {
+
+        int bind(sqlite3_stmt* stmt, int index, const std::chrono::sys_days& value) const {
+            return statement_binder<std::string>().bind(stmt, index, sysDaysToString(value));
+        }
+    };
+
+    /**
+	 *  field_printer is used in `dump` and `where` functions. Here we have to create
+	 *  a string from mapped object.
+	 */
+    template<>
+    struct field_printer<std::chrono::sys_days> {
+        std::string operator()(const std::chrono::sys_days& t) const {
+            return sysDaysToString(t);
+        }
+    };
+
+    /**
+	 *  This is a reverse operation: here we have to specify a way to transform string received from
+	 *  database to our sysdays object. Here we call `SysDaysFromString` and throw `std::runtime_error` if it returns
+	 *  null_sys_day. Every `row_extractor` specialization must have `extract(const char*)`, `extract(sqlite3_stmt *stmt, int columnIndex)`
+	 *	and `extract(sqlite3_value* value)`
+	 *  functions which return a mapped type value.
+	 */
+    template<>
+    struct row_extractor<std::chrono::sys_days> {
+        std::chrono::sys_days extract(const char* row_value) const {
+            auto sd = sysDaysFromString(row_value);
+            if(sd != null_sys_day) {
+                return sd;
+            } else {
+                throw std::runtime_error("incorrect date string (" + std::string(row_value) + ")");
+            }
+        }
+
+        std::chrono::sys_days extract(sqlite3_stmt* stmt, int columnIndex) const {
+            auto str = sqlite3_column_text(stmt, columnIndex);
+            return this->extract((const char*)str);
+        }
+        std::chrono::sys_days extract(sqlite3_value* row_value) const {
+            auto characters = reinterpret_cast<const char*>(sqlite3_value_text(row_value));
```

Comment:
Rather use a C cast here just like above and all other places concerning row_extractor.

Reply:
why a C cast? MSVC does not allow a static_cast<> because the input is unsigned and the output signed that's why I had to use reinterpret_cast<> -- I always thought C++ casts should be preferred over C casts, am I wrong?

Reply:
3 lines up, you have a C cast, just like all the other line extractors use. sqlite_orm uses C casts or value casts throughout, so I would just be consistent.
As for preferring reinterpret_cast over C casts - I'm not in favor of always using reinterpret_cast (and often not even static_cast) simply because it's less readable. And a C cast is perfectly valid. Just my 2 cents :)